code/modules/power/heavy_emitter.dm

```diff
@@ -0,0 +1,261 @@
+		add_load(active_power_usage)
+		radiation_pulse(src,500,can_contaminate=FALSE)
+		visible_message("<span class='notice'>Heavy Emitter Core is powering the cannon....</span>")
+		INVOKE_ASYNC(linked_cannon,/obj/machinery/power/heavy_emitter/cannon.proc/fire)
+		heat += 500
+
+	if(heat > max_heat)
+		explosion(src,4,8,16)
+		return
+
+	for(var/V in vents)
+		if(heat <= 0 || !V)
+			break
+		var/obj/machinery/power/heavy_emitter/vent/vent = V
+		if(vent.vent_gas())
+			heat -= 100
+
+
+/obj/machinery/power/heavy_emitter/arm
+	name = "Seismic Stabilizer Arm"
+	desc = "Dampens the recoil from firing to virtually nothing"
+	icon_state = "arm"
+
+/obj/machinery/power/heavy_emitter/interface
+	name = "Kinetic Amplification Manipulation Interface"
+	desc = "Allows for control over the core."
+	icon_state = "interface_off"
+	///Core connected to this thing
+	var/obj/machinery/power/heavy_emitter/centre/connected_core
+
+/obj/machinery/power/heavy_emitter/interface/attack_hand(mob/living/user)
+	. = ..()
+	var/turf/T = get_step(src,turn(dir,180))
+	var/obj/machinery/power/heavy_emitter/centre/centre = locate() in T
+
+	if(!centre || !centre.check_part_connectivity())
+		turn_off()
+		return
+
+	connected_core = centre
+
+	if(connected_core.firing)
+		to_chat(user, "<span class='warning'>You power on the Heavy Emitter!</span>")
+		turn_off()
+	else
+		to_chat(user, "<span class='warning'>You disable the Heavy Emitter!</span>")
+		turn_on()
+
+
+/obj/machinery/power/heavy_emitter/interface/turn_on()
+	icon_state = "interface"
+	connected_core.turn_on()
+
+/obj/machinery/power/heavy_emitter/interface/turn_off()
+	icon_state = "interface_off"
+	connected_core.turn_off()
+
+/obj/machinery/power/heavy_emitter/vent
+	name = "Energy Core Vent"
+	desc = "Circulates air around the core, preventing it from overheating."
+	icon_state = "vent"
+
+/obj/machinery/power/heavy_emitter/vent/proc/vent_gas()
+	var/turf/open/open_turf = get_step(src,dir)
+	if(!istype(open_turf))
+		return FALSE
+	var/datum/gas_mixture/gases = open_turf.return_air()
+	//Space can exist
+	if(gases)
+		gases.temperature += 100
+	flick("vent_on",src)
+	return TRUE
+
+/obj/machinery/power/heavy_emitter/cannon
+	name = "Energy Optic Converging Cannon"
+	desc = "Converges the energy from the core into a singular destructive beam."
+	icon_state = "cannon"
+	var/warmup_sound = 'sound/machines/warmup1.ogg'
+	var/cooldown_sound = 'sound/machines/cooldown1.ogg'
+	var/projectile_sound = 'sound/weapons/beam_sniper.ogg'
+	var/projectile_type = /obj/projectile/beam/emitter/heavy
+
+/obj/machinery/power/heavy_emitter/cannon/proc/fire()
+	playsound(src, warmup_sound, 100)
+	sleep(5 SECONDS)
+	var/turf/hot_turf = get_step(src,dir)
+	new /obj/effect/hotspot(hot_turf)
+	var/obj/projectile/proj = new projectile_type(get_turf(src))
+	playsound(src, projectile_sound, 100, TRUE)
+	proj.firer = src
+	proj.fired_from = src
+	proj.fire(dir2angle(dir))
+	playsound(src, cooldown_sound, 100)
+	sleep(2 SECONDS)
```

Comment:
Why?

Reply:
I'm assuming this was discussed over discord, could you put it here too for posterity?

Reply:
no this was just a mistake, no need to sleep at the end of the proc